File: Lab/SEM4/MPP/lab7/moto_java/common/src/main/java/org/zdroba/sync/Response.java
package org.zdroba.sync;

import java.io.Serializable;

public class Response implements Serializable {
    private final ResponseType type;
    private final String requestType;
    private final String message;

    public Response(ResponseType type, String requestType, String message) {
        this.type = type;
        this.requestType = requestType;
        this.message = message;
    }

    public ResponseType getType() {
        return type;
    }

    public String getRequestType() {
        return requestType;
    }

    public String getMessage() {
        return message;
    }

    @Override
    public String toString() {
        return "RESPONSE:" + type + "|" + requestType + "|" + (message != null ? message : "");
    }

    public static Response fromString(String str) {
        if (!str.startsWith("RESPONSE:"))
            throw new IllegalArgumentException("Not a response string");

        String[] parts = str.substring(9).split("\\|", 3);
        ResponseType type = ResponseType.valueOf(parts[0]);
        String requestType = parts[1];
        String message = parts.length > 2 ? parts[2] : null;

        return new Response(type, requestType, message);
    }
}

File: Lab/SEM4/MPP/lab9/moto_c/src/Common/sync/Response.cs
namespace moto_c.Common.sync;

public class Response
{
    public ResponseType type;
    public string requestType;
    public string message;

    public Response(ResponseType type, string requestType, string message)
    {
        this.type = type;
        this.requestType = requestType;
        this.message = message;
    }

    public ResponseType GetType()
    {
        return type;
    }

    public string GetRequestType()
    {
        return requestType;
    }

    public string GetMessage()
    {
        return message;
    }

    public override string ToString()
    {
        return "RESPONSE:" + type + "|" + requestType + "|" + (message ?? "");
    }

    public static Response FromString(string str)
    {
        if (!str.StartsWith("RESPONSE:"))
            throw new ArgumentException("Not a response string");

        string[] parts = str.Substring(9).Split('|', 3);

        ResponseType type = Enum.Parse<ResponseType>(parts[0]);
        string requestType = parts[1];
        string message = parts.Length > 2 ? parts[2] : null;

        return new Response(type, requestType, message);
    }
}